lib/frrstr.c

```diff
@@ -78,6 +78,13 @@ char *frrstr_join(const char **parts, int argc, const char *join)
 	size_t len = 0;
 	size_t joinlen = join ? strlen(join) : 0;
 
+	if (!argc) {
+		str = p = XMALLOC(MTYPE_TMP, 1);
+		if (!str)
```

Comment:
`XMALLOC` will never fail, this case is not necessary. Please rewrite this to
```
if (!argc)
    return NULL;
```
and remove the labels.

Reply:
Ok, thank you.